packages/compass-sidebar/src/components/multiple-connections/sidebar.tsx

```diff
@@ -240,20 +245,32 @@ export function MultipleConnectionSidebar({
               }
             }}
             initialConnectionInfo={editingConnectionInfo}
-            onSaveAndConnectClicked={(connectionInfo) => {
-              void connect(connectionInfo);
-            }}
-            onSaveClicked={(connectionInfo) => {
-              return saveEditedConnection(connectionInfo);
-            }}
-            onCancel={() => {
-              cancelEditConnection(editingConnectionInfo.id);
-            }}
             connectionErrorMessage={
               connectionErrors[editingConnectionInfo.id]?.message
             }
             openSettingsModal={openSettingsModal}
             {...formPreferences}
+            onCancel={() => {
+              cancelEditConnection(editingConnectionInfo.id);
+            }}
+            onSaveClicked={(connectionInfo) => {
+              return saveEditedConnection(connectionInfo);
+            }}
+            onConnectClicked={
+              isEditingNewConnection || disableEditingConnectedConnection
+                ? undefined
+                : (connectionInfo) => {
+                    void connect(connectionInfo);
+                  }
+            }
+            onSaveAndConnectClicked={
+              disableEditingConnectedConnection
```

Comment:
I changed all these to not include the button callback prop if it shouldn't show up. That makes the logic easier and less error prone because you don't have to check both the boolean AND the button to know which buttons to include inside the form and to know which one to call. 

`disableEditingConnectedConnection` is then only used when a button isn't directly involved like when showing the banner.